crypto/schemes.go

```diff
@@ -0,0 +1,224 @@
+package crypto
+
+import (
+	"crypto/cipher"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"os"
+
+	"github.com/drand/kyber"
+
+	bls "github.com/drand/kyber-bls12381"
+	"github.com/drand/kyber/pairing"
+	"github.com/drand/kyber/sign"
+
+	// The package github.com/drand/kyber/sign/bls is deprecated because it is vulnerable to
+	// rogue public-key attack against BLS aggregated signature. The new version of the protocol can be used to
+	// make sure a signature aggregate cannot be verified by a forged key. You can find the protocol in kyber/sign/bdn.
+	// Note that only the aggregation is broken by the attack and a later version will merge bls and asmbls.
+	// The way we are using this package does not do any aggregation and we're only using simple signatures and thus
+	// this is not a security issue for drand.
+	//nolint:staticcheck
+	signBls "github.com/drand/kyber/sign/bls"
+	"github.com/drand/kyber/sign/schnorr"
+	"github.com/drand/kyber/sign/tbls"
+
+	"github.com/drand/kyber/util/random"
+	"golang.org/x/crypto/blake2b"
+
+	"hash"
+)
+
+type hashableBeacon interface {
+	GetPreviousSignature() []byte
+	GetRound() uint64
+}
+
+type signedBeacon interface {
+	hashableBeacon
+	GetSignature() []byte
+}
+
+type Scheme struct {
```

Comment:
a small comment indicating what Scheme is for ? or at the top level ? so at least we see something on godoc ?

Reply:
Good point, thanks